### PR TITLE
fix(anchor): allow layout and space props

### PR DESCRIPTION
**Pull request by richbachman**
- [x] Added all layout and space props to anchor
- [x] Added tests
- [x] Added margin and padding shorthand props to style-props

### CODE REVIEW

File: packages/paste-style-props/src/types/space.ts [hunk 1]
```diff
@@ -11,22 +11,36 @@ export type Margin = Space | 'auto';
...
   paddingRight?: Padding;
+  pr?: Padding;
   paddingBottom?: Padding;
+  pb?: Padding;
   paddingLeft?: Padding;
+  pl?: Padding;
   paddingX?: Padding;
+  px?: Padding;
   paddingY?: Padding;
+  py?: Padding;
```
[the remark this answers is not in the document]
Comment (richbachman) [reply packages/paste-style-props/src/types/space.ts:43]: Nope. They weren't working. Noticed they were in the `MARGIN_PROPS` const, but not the interface `MarginProps`.

[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-style-props/src/types/space.ts:43]: Oh no wait, we're good. We were only allowing our props from our package, so i actually think we're good in only allowing what's defined in here already as our `LayoutProps`.

this was the original import into Anchor.

`import {LayoutProps, SpaceProps} from '@twilio-paste/style-props';`

Phew! We can just add margin/padding top, bottom, left, right, x and y to Anchor in that case.

[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-style-props/src/types/space.ts:43]: `pr` would have worked if you were using plain js but it wasn't an intended API so we are covered.

[the remark this answers is not in the document]
Comment (richbachman) [reply packages/paste-style-props/src/types/space.ts:43]: So I can drop the `p, pr, m, mr, etc` from the space types and just use the spelled out versions?

[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-style-props/src/types/space.ts:43]: Yeah, just put this back to the way it was, and only provide whats listed.

[the remark this answers is not in the document]
Comment (richbachman) [reply packages/paste-style-props/src/types/space.ts]: Done.

**Review by SiTaggart — APPROVED** [review]

**Pull request merged**